File: Python/largeAirportsDF.py
from pyspark.sql import SparkSession, functions


def main() -> None:
    spark_session = SparkSession\
        .builder\
        .master("local[4]") \
        .appName("Large airports (Dataframe)") \
        .getOrCreate()

    logger = spark_session._jvm.org.apache.log4j
    logger.LogManager.getLogger("org").setLevel(logger.Level.WARN)

    df_airports = spark_session \
        .read \
        .options(header='true', inferschema='true') \
        .option("delimiter", ",") \
        .csv("data/airports.csv") \
        .persist()

    df_countries = spark_session \
        .read \
        .options(header='true', inferschema='true') \
        .option("delimiter", ",") \
        .csv("data/countries.csv") \
        .persist()

    df_airports.printSchema()
    df_airports.show()

    df_airports\
        .filter(df_airports["type"] == "large_airport")\
        .drop("name")\
        .join(df_countries, df_airports["iso_country"] == df_countries["code"]) \
        .groupBy("name")\
        .count()\
        .sort("count", ascending=False)\
        .show(10)


if __name__ == '__main__':
    main()

File: JavaM9/src/main/java/org/mbd/m9/LargeAirportsDF.java
package org.mbd.m9;

import org.apache.log4j.Level;
import org.apache.log4j.Logger;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

import static org.apache.spark.sql.functions.col;

public class LargeAirportsDF {
    public static void main(String[] args) {

        Logger.getLogger("org").setLevel(Level.OFF);

        SparkSession sparkSession =
                SparkSession.builder()
                        .appName("Large airports (Dataframe)")
                        .master("local[4]")
                        .getOrCreate();

        Dataset<Row> dfAirports = sparkSession
                .read()
                .format("csv")
                .option("header", "true")
                .option("inferSchema", "true")
                .load("data/airports.csv")
                .cache();

        dfAirports.printSchema();
        dfAirports.show(10);

        Dataset<Row> dfCountries = sparkSession
                .read()
                .format("csv")
                .option("header", "true")
                .option("inferSchema", "true")
                .load("data/countries.csv")
                .cache();

        dfCountries.printSchema();
        dfCountries.show(10);

        dfAirports
                .filter(col("type").equalTo("large_airport"))
                .drop(col("name"))
                .join(dfCountries, dfAirports.col("iso_country").equalTo(dfCountries.col("code")))
                .groupBy("name")
                .count()
                .sort(col("count").desc())
                .show(10);

        sparkSession.stop();
    }
}
